src/librustc_target/spec/mod.rs

```diff
@@ -373,6 +373,8 @@ supported_targets! {
     ("armv7-unknown-cloudabi-eabihf", armv7_unknown_cloudabi_eabihf),
     ("i686-unknown-cloudabi", i686_unknown_cloudabi),
     ("x86_64-unknown-cloudabi", x86_64_unknown_cloudabi),
+
+    ("riscv32imac-unknown-none", riscv32imac_unknown_none),
```

Comment:
Perhaps the last part of the triple should be `unknown-elf` to match the triple used by the GCC tooling?

Reply:
I don't know if this is necessary. If it where something other than elf we'd have riscv32imac-unknown-(darwin|msvc).

Reply:
We'll be adding `riscv64imac-linux-gnu` in the future. I'd like to see some consistency in the triple format: `$ARCH_PLUS_INSN_SET-$OS-$ENV_OR_FORMAT`. `*-unknown-none` sounds like either the environment or the object format is `none` and that's not the case.

Reply:
I think the `unknown` refers to the cpu vendor, in case there are vendor specific bugs/optimizations/extensions. So is `riscv32imac-unknown-none-elf` ok?

Reply:
Sounds like the most future proof choice. :+1: